### ISSUE TITLE
Podman machine perpetually in MachineState Starting after podman machine stop

### ISSUE THREAD

**Issue by Aenimus** [containers/podman#18666]
### Issue Description

After running `podman machine stop`, despite a success signal, `podman machine info` perpetually claims that the MachineState is Starting. `podman machine start` always fails.

### Steps to reproduce the issue

1. `podman machine stop` -> Machine "podman-machine-default" stopped successfully
2. `podman machine start` -> Error: cannot start VM podman-machine-default: VM already running or starting
3. `podman machine info` -> MachineState is perpetually Starting; subsequent stops claim success but change nothing
4. restart laptop
5. repeat

I was forced to use `podman machine rm` to remove the machine.

### Describe the results you received

`podman stop` -> Machine "podman-machine-default" stopped successfully
`podman machine start` -> Error: cannot start VM podman-machine-default: VM already running or starting
`podman machine info` -> MachineState: Starting

The MachineState never changes from Starting. This persists through laptop restart.

### Describe the results you expected

The machine to stop.

### podman info output

```yaml
Version:
  APIVersion: 4.5.0
  Built: 1681478900
  BuiltTime: Fri Apr 14 14:28:20 2023
  GitCommit: 75e3c12579d391b81d871fd1cded6cf0d043550a
  GoVersion: go1.20.3
  Os: darwin
  OsArch: darwin/arm64
  Version: 4.5.0

I can output the rest of info if it's useful, but I've since removed the perpetually starting machine and created a new one.
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

Yes

### Additional environment details

M1 max macOS Ventura 13.3.1 (a)
Installed with brew

### Additional information

Exactly this closed issue: https://github.com/containers/podman/issues/16815

Someone else had the same problem with the last few days.

**Comment by Aenimus** [comment]
This issue might also be identical to https://github.com/containers/podman/issues/18662
I apologise for not checking existing issues first; I just saw the other issue I linked was closed and created a new one.

An update:

I cannot replicate the behaviour on the new machine; it stops successfully. However, `podman machine stop` on this new machine outputs "waiting for VM to exit"; output I don't believe I received when it became stuck restarting.

It may also be worth noting that the podman machine was created around 4 months ago, and I had to stop and start it in the past (successfully).
